source_handlers_test.go

```diff
@@ -1420,6 +1420,44 @@ func TestSuperKeyDestroyWithOwnershipWhenUserIsNotAllowedToDelete(t *testing.T)
 	}
 }
 
+func TestSourceDeleteWithOwnership(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+	testutils.SkipIfNotSecretStoreDatabase(t)
+
+	accountNumber := "112567"
+	userIDWithOwnRecords := "user_based_user"
+
+	applicationTypeID := fixtures.TestApplicationTypeData[3].Id
+	sourceTypeID := fixtures.TestSourceTypeData[2].Id
+	recordsWithUserID, user, err := dao.CreateSourceWithSubResources(sourceTypeID, applicationTypeID, accountNumber, &userIDWithOwnRecords)
+	if err != nil {
+		t.Errorf("unable to create source: %v", err)
+	}
+
+	src := recordsWithUserID.Sources[0]
+	tenantID := src.TenantID
+
+	id := fmt.Sprintf("%d", src.ID)
+	c, _ := request.CreateTestContext(
+		http.MethodDelete,
+		"/api/sources/v3.1/sources/"+id,
+		nil,
+		map[string]interface{}{
+			"tenantID": tenantID,
+			"userID":   user.Id,
+		},
+	)
+
+	c.SetParamNames("id")
+	c.SetParamValues(id)
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+
+	err = SourceDelete(c)
+	if err != nil {
+		t.Error(err)
+	}
```

Comment:
just thinking 🤔 in this case new user was created for this test? so after deletion of the source the expected result it that the source and all subresources are deleted but the user is still in database? if this test creates new user, maybe you should delete this user in the end of the test ... but maybe it is not needed ...but I'm just thinking out loud 😇